ciao-cli/client.go

```diff
@@ -286,47 +281,86 @@ func (client *Client) checkPrivilege() bool {
 	return false
 }
 
-// ListEvents retrieves the events for either all or the desired tenant
-func (client *Client) ListEvents(tenantID string) (types.CiaoEvents, error) {
-	var events types.CiaoEvents
-	var url string
+func (client *Client) getResource(url string, content string, query []queryValue, result interface{}) error {
+	resp, err := client.sendHTTPRequest("GET", url, query, nil, content)
+	if err != nil {
+		return errors.Wrapf(err, "Error making HTTP request to %s", url)
+	}
+	defer resp.Body.Close()
```

Comment:
resp.Body is going to get closed twice, once when this function exits and once when unmarshalHTTPResponse exits.   This behaviour is undefined.

Reply:
I didn't realise unmarshalHTTPResponse() did that. I think it's bad style to i've made sure that wherever a response is generated the code there is responsible for cleaning up.